### ISSUE TITLE
[🐛 BUG] Setting width/height with CSS units not working for metrics

### ISSUE THREAD

**Issue by FlorianJacta** [Avaiga/taipy#2141]
### What went wrong? 🤔

Setting width and height for metrics with CSS units is not working. This won't change the width/height as expected.

This works when just putting a number. It took a while for me to figure out (I went looking at the actual PR).

If this issue could be backported to another patch, this would be great.

### Expected Behavior

This should work with all CSS units supported by Taipy. Or the documentation should change to reflect this behavior.

### Steps to Reproduce Issue

```python
from taipy.gui import Gui

value = 50

page = """
<|{value}|metric|>

## Works
<|{value}|metric|width=300|height=300|>

## Height not working, Width not working
<|{value}|metric|width=150px|height=300px|>
<|{value}|metric|width=300px|height=300px|>


## Height not working, Width not working
<|{value}|metric|width=150rem|height=150rem|>
<|{value}|metric|width=300rem|height=150rem|>
"""

Gui(page).run()
```

### Screenshots

![image](https://github.com/user-attachments/assets/6a8fb469-27af-466c-a9bd-f4cd75210224)

### Version of Taipy

4.0

### Acceptance Criteria

- [ ] A unit test reproducing the bug is added.
- [ ] Any new code is covered by a unit tested.
- [ ] Check code coverage is at least 90%.
- [ ] The bug reporter validated the fix.
- [ ] Related issue(s) in taipy-doc are created for documentation and Release Notes are updated.

### Code of Conduct

- [X] I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+).
- [ ] I am willing to work on this issue (optional)

**Comment by nehaaprasad** [comment]
Hello @FlorianJacta :)

* I think update Dimension Handling: Allow width and height to accept both numerical values and CSS units (e.g., px, rem).
* Dimension Parsing: Implement logic to recognize CSS unit strings or convert numeric values to pixel-based strings (e.g., 150 becomes "150px").

**Comment by FlorianJacta** [comment]
I think 150 and 150px are two different units. It is related to this issue too: https://github.com/Avaiga/taipy/issues/2142 so you should also be assigned there

**Comment by vgauraha62** [comment]
Parameters in the 'metric' component do not accept units like, px or rem.
Use numerical values without units (e.g. width=150, height=300).
This should apply the desired dimensions to the metric component.

**Comment by Yubarajhalder** [comment]
The issue lies in how you're setting the width and height properties in the metric widget.

Taipy GUI uses a specific syntax for styling, and width and height should be defined within a style attribute.

Try modifying your page string as follows:


page = """
<|{value}|metric|style={"width": "300px", "height": "300px"}|>
"""


Or, for multiple styles:


page = """
<|{value}|metric|style={"width": "300px", "height": "300px", "font-size": "24px"}|>
"""


This should correctly apply the width and height styles to your metric widget.

Additionally, consider using units like px for pixels or % for percentage-based sizing.

For more information on Taipy GUI styling, refer to the official documentation:

**Comment by FlorianJacta** [comment]
@Yubarajhalder your answer is not answering the issue

**Comment by philliparaujo** [comment]
Hi, could you please assign me to this issue?

**Comment by FlorianJacta** [comment]
Sure @philliparaujo, I have assigned you to this issue!

**Comment by FlorianJacta** [comment]
This doesn't seem to be fixed in 4.0.2.dev1 and we also have by default a metric taking the entire screen. However, this is fixed in develop

**Comment by AlexandreSajus** [comment]
Fixed in 4.0.2.dev2!
